Zadania2/prosteZadanka2_4.py

```diff
@@ -0,0 +1,79 @@
+class Student:
+    first_name = None
+    last_name = None
+
+    def __init__(self, first_name, last_name):
+        self.first_name = first_name
+        self.last_name = last_name
+
+
+class KlassFullException(Exception):
+    pass
+
+class Klasa:
+    student_list = []
+
+    def __init__(self, max_size):
+        if isinstance(max_size, int) and max_size > 0:
+            self.max_size = max_size
+        else:
+            print("Coś poszło nie tak.")
```

Comment:
dobrze jest jakimś wyjątkiem sypnać bo inaczej obiektu stworzonego można używać w niezamierzony sposób np. jak nie ustawimy max_size to użycie metody add_student zakończy się przewróceniem dobiero aplikacji.

Reply:
Rozumiem. Czy dla takiego przypadku można robić walidację danych w metodzie __new__, która de facto tworzy obiekt? Wtedy zmienna będzie odwoływać się do None i żadnej metody nie wywołamy. Czy też lepiej unikać grzebania z __new__ ?